### PR TITLE
Feature/vps poc task

### PR DESCRIPTION
**Pull request by whispyy**

### CODE REVIEW

File: client/app/vps/additional-disk/vps-additional-disk.controller.js [hunk 1]
```diff
@@ -23,16 +23,14 @@ class VpsAdditionalDiskCtrl {
     hasAdditionalDisk () {
         this.loaders.init = true;
         this.VpsService.hasAdditionalDiskOption(this.serviceName)
-            .then(() => { this.hasAdditionalDiskOption = true })
-            .catch(() => {
-                this.hasAdditionalDiskOption = false;
-            })
-            .finally(() => {
-                this.loaders.init = false;
+            .then(() => {
```
Comment (whispyy) [inline client/app/vps/additional-disk/vps-additional-disk.controller.js]: this is to solve @Mboulianne previous comments :)

File: client/app/vps/veeam/vps-veeam.html [hunk 2]
```diff
@@ -7,6 +7,7 @@
         <div data-ng-if="$ctrl.veeam.state !== 'disabled'">
             <p data-ng-if="!$ctrl.veeam.accessInfos.restorePoint" data-translate="vps_tab_VEEAM_dashboard_no_current_restore_point"></p>
             <dl data-ng-if="$ctrl.veeam.accessInfos.restorePoint">
+                <p data-translate="vps_tab_VEEAM_dashboard_current_restore_point"></p>
```
Comment (JDownloader) [inline client/app/vps/veeam/vps-veeam.html]: un <p> dans un <dl> c'est très étrange, on devrait se limiter à <dt> et <dd>

File: client/app/vps/vps-detail.controller.js [hunk 3]
```diff
@@ -17,6 +18,7 @@ class VpsDetailCtrl {
                 serviceName: this.serviceName
             }
         });
+        this.VpsTaskService.subscribe(this.serviceName);
```
Comment (JDownloader) [inline client/app/vps/vps-detail.controller.js]: is-it normal to not have any unsubscribe?

File: client/app/vps/vps-task.service.js [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+        return this.$http.get(["/sws/vps", serviceName,"info"].join("/"), {serviceType: "aapi"})
+            .then(result => result.data)
+            .catch(err => this.$q.reject(err));
+    }
+
+    getTaskInProgress (serviceName, type) {
+        var result = null;
+        return this.getSelectedVps(serviceName).then(vps => {
+            if (vps && vps.name) {
+                return this.$http.get(["/sws/vps", vps.name, "tasks/uncompleted"].join("/"), {
```
Comment (JDownloader) [inline client/app/vps/vps-task.service.js]: this could be done in a separate function or service

File: client/app/vps/vps-task.service.js [hunk 5]
```diff
@@ -0,0 +1,138 @@
+class VpsTaskService {
```
Comment (JDownloader) [inline client/app/vps/vps-task.service.js]: I like that you have created a new service for this purpose :-)

File: client/app/vps/vps-task.service.js [hunk 6]
```diff
@@ -0,0 +1,138 @@
...
+            }, 5000);
+        }
+    }
+
+    /*
+     * manageMessage : manage message to display
+     *
+     */
+    manageMessage (task) {
+        if (task.progress !== 100) {
```
Comment (JDownloader) [inline client/app/vps/vps-task.service.js]: 100 seams like a magic number, using a constant would give meaning

Comment (whispyy) [reply client/app/vps/vps-task.service.js]: Yes Thanks.

File: client/app/vps/vps-task.service.js (hunk 4 above)
Comment (whispyy) [reply client/app/vps/vps-task.service.js]: Done

File: client/app/vps/veeam/vps-veeam.html (hunk 2 above)
Comment (whispyy) [reply client/app/vps/veeam/vps-veeam.html]: done

File: client/app/vps/vps-detail.controller.js (hunk 3 above)
Comment (Mboulianne) [reply client/app/vps/vps-detail.controller.js]: I agree this is weird.  You should have an unsubscribe function.

File: client/app/vps/vps-task.service.js [hunk 7]
```diff
@@ -0,0 +1,126 @@
...
+        }, "iaas.vps.detail");
+    }
+
+    /*
+     * flushMessages : flush all messages that have 'task' class
+     *
+     */
+    flushMessages () {
+        _.forEach(this.CloudMessage.getMessages("iaas.vps.detail"), message => {
+            if (message.class == "task") {
```
Comment (Mboulianne) [inline client/app/vps/vps-task.service.js]: === :P

File: client/app/vps/vps-task.service.js [hunk 8]
```diff
@@ -0,0 +1,126 @@
...
+                message.dismissed = true;
```
Comment (Mboulianne) [inline client/app/vps/vps-task.service.js]: to be safe you should call message.dismiss() or message.onDismiss (can't remember the correct name)

File: client/app/vps/vps-task.service.js [hunk 9]
```diff
@@ -0,0 +1,126 @@
...
+     *
+     */
+    createMessage (task) {
+        this.CloudMessage.warning({
+            id: task.id,
+            class: "task",
+            title: this.messageType(task.type),
+            textHtml: this.template(task.type, task.progress),
+            progress: task.progress
+        }, "iaas.vps.detail");
```
Comment (Mboulianne) [inline client/app/vps/vps-task.service.js]: I'd rather not hardcode a state for that and change the logic onStateChange of CloudMessage to not flush task message on state change.

File: client/app/vps/vps-task.service.js [hunk 10]
```diff
@@ -0,0 +1,126 @@
...
+        });
+
+        if (this.firstCall && !_.isEmpty(tasks)) {
+            this.firstCall = false;
+            this.$rootScope.$broadcast("tasks.pending");
+        }
+
+        // refresh while there's pending task
+        if (!this.firstCall) {
+            this.$timeout(() => {
```
Comment (Mboulianne) [inline client/app/vps/vps-task.service.js]: interval + unsubscribe function that kill the interval if needs be

File: client/app/vps/vps-task.service.js [hunk 11]
```diff
@@ -0,0 +1,126 @@
...
+    constructor ($http, $q, $rootScope, $timeout, $translate, CloudMessage) {
+        this.$http = $http;
+        this.$q = $q;
+        this.$rootScope = $rootScope;
+
+        this.$timeout = $timeout;
+        this.$translate = $translate;
+        this.CloudMessage = CloudMessage;
+
+        this.firstCall = true;
```
Comment (Mboulianne) [inline client/app/vps/vps-task.service.js]: I don't like this thing but I understand you have no choice :(

File: client/app/vps/vps-task.service.js [hunk 12]
```diff
@@ -0,0 +1,126 @@
...
+            this.manageMessage(task);
+        });
+
+        if (this.firstCall && !_.isEmpty(tasks)) {
+            this.firstCall = false;
+            this.$rootScope.$broadcast("tasks.pending");
+        }
+
+        // refresh while there's pending task
+        if (!this.firstCall) {
```
Comment (JDownloader) [inline client/app/vps/vps-task.service.js]: This if condition seams unuseful, you assign false at line 61

File: client/app/vps/vps-task.service.js [hunk 13]
```diff
@@ -0,0 +1,126 @@
...
+            this.$timeout(() => {
+                if (!_.isEmpty(tasks)) {
+                    this.$rootScope.$broadcast("tasks.pending");
+                    this.getTasks(serviceName);
+                } else {
+                    this.flushMessages();
+                    this.$rootScope.$broadcast("tasks.success");
+                    this.CloudMessage.success(this.$translate.instant("vps_dashboard_task_finish"));
+                }
+            }, 5000);
```
Comment (JDownloader) [inline client/app/vps/vps-task.service.js]: magic number are magic

File: client/app/vps/vps-task.service.js (hunk 11 above)
Comment (whispyy) [reply client/app/vps/vps-task.service.js]: Yup. In fact I didn't want to enter $timeout the firstTime if there's no pending task. BUT I want to enter it lastTime to return msg and broadcast it is finish.

File: client/app/vps/vps-task.service.js (hunk 8 above)
Comment (whispyy) [reply client/app/vps/vps-task.service.js]: none of them works..

File: client/app/vps/vps-task.service.js (hunk 10 above)
Comment (whispyy) [reply client/app/vps/vps-task.service.js]: Done :)

File: client/app/vps/vps-task.service.js [hunk 14]
```diff
@@ -73,7 +89,7 @@ class VpsTaskService {
                     this.$rootScope.$broadcast("tasks.success");
                     this.CloudMessage.success(this.$translate.instant("vps_dashboard_task_finish"));
                 }
-            }, 5000);
+            }, this.TIMER);
```
Comment (JDownloader) [inline client/app/vps/vps-task.service.js]: this looks more le a time_out to me

**Comment by whispyy** [comment]
I've just finished a complete refacto of vps-task-services to use OvhPolls services which it make this services a bit lighter.

**Pull request merged**